[mayaa-user:1087] 独自プロセッサーを使うとエラーが発生する場合がある

下記のもので a.html -> b.html の順で実行すると `ClassCastException` が発生する。

sample.mld

```
    <processor name="formatDate" class="com.example.FormatDateProcessor">
        <property name="value" required="true" expectedClass="java.time.LocalDateTime"/>
        <property name="default" expectedClass="java.lang.String"/>
        <property name="pattern" expectedClass="java.lang.String"/>
    </processor>
```

a.mayaa

```
  <m:beforeRender><![CDATA[
    var value = "TEST";
  ]]></m:beforeRender>
 
  <m:echo m:id="item">
    <m:attribute name="value" value="${value}" />
  </m:echo>
```

b.mayaa

```
  <m:beforeRender><![CDATA[
    var value = java.time.LocalDateTime.now();
  ]]></m:beforeRender>
 
  <a:formatDate m:id="item" value="${value}" pattern="yyyy-MM-dd" />
```

例外

```
java.lang.ClassCastException: Cannot cast java.lang.String to java.time.LocalDateTime
        at java.lang.Class.cast(Class.java:3369)
        at com.example.FormatDateProcessor.format(FormatDateProcessor.java:45)
        at com.example.FormatDateProcessor.doStartProcess(FormatDateProcessor.java:33)
        at org.seasar.mayaa.impl.engine.RenderUtil.renderTemplateProcessor(RenderUtil.java:134)
        at org.seasar.mayaa.impl.engine.RenderUtil.renderProcessorTree(RenderUtil.java:265)
        at org.seasar.mayaa.impl.engine.TemplateImpl.doTemplateRender(TemplateImpl.java:202)
        at org.seasar.mayaa.impl.engine.PageImpl.renderTemplate(PageImpl.java:241)
        at org.seasar.mayaa.impl.engine.RenderUtil.renderPage(RenderUtil.java:349)
        at org.seasar.mayaa.impl.engine.PageImpl.doPageRender(PageImpl.java:189)
        at org.seasar.mayaa.impl.engine.EngineImpl.doPageService(EngineImpl.java:373)
        at org.seasar.mayaa.impl.engine.EngineImpl.doService(EngineImpl.java:481)
        at org.seasar.mayaa.impl.MayaaServlet.doService(MayaaServlet.java:107)
        at org.seasar.mayaa.impl.MayaaServlet.doGet(MayaaServlet.java:90)
 ```

再現コードを作成したいが発生する現象が異なる。
Class.cast でなく commons-beanutils が使われている。発生パターンが異なるのか。
```
java.lang.IllegalArgumentException: Cannot invoke org.seasar.mayaa.regressions.issue13.FormatDateProcessor.setValue on bean class 'class org.seasar.mayaa.regressions.issue13.FormatDateProcessor' - argument type mismatch - had objects of type "java.lang.String" but expected signature "java.time.LocalDateTime"
	at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2235)
	at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2151)
	at org.apache.commons.beanutils.PropertyUtilsBean.setNestedProperty(PropertyUtilsBean.java:1957)
	at org.apache.commons.beanutils.PropertyUtilsBean.setProperty(PropertyUtilsBean.java:2064)
	at org.apache.commons.beanutils.PropertyUtils.setProperty(PropertyUtils.java:858)
	at org.seasar.mayaa.impl.util.ObjectUtil.setProperty(ObjectUtil.java:297)
	at org.seasar.mayaa.impl.builder.library.ProcessorDefinitionImpl.settingProperty(ProcessorDefinitionImpl.java:127)
	at org.seasar.mayaa.impl.builder.library.ProcessorDefinitionImpl.settingPropertySet(ProcessorDefinitionImpl.java:172)
	at org.seasar.mayaa.impl.builder.library.ProcessorDefinitionImpl.createTemplateProcessor(ProcessorDefinitionImpl.java:212)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.createProcessor(TemplateBuilderImpl.java:391)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.resolveInjectedNode(TemplateBuilderImpl.java:410)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.walkParsedTree(TemplateBuilderImpl.java:565)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.walkParsedTree(TemplateBuilderImpl.java:569)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.walkParsedTree(TemplateBuilderImpl.java:569)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.doInjection(TemplateBuilderImpl.java:645)
	at org.seasar.mayaa.impl.builder.TemplateBuilderImpl.afterBuild(TemplateBuilderImpl.java:191)
	at org.seasar.mayaa.impl.builder.SpecificationBuilderImpl.build(SpecificationBuilderImpl.java:79)
	at org.seasar.mayaa.impl.engine.specification.SpecificationImpl.build(SpecificationImpl.java:154)
	at org.seasar.mayaa.impl.engine.TemplateImpl.build(TemplateImpl.java:225)
	at org.seasar.mayaa.impl.engine.EngineImpl.createSpecificationInstance(EngineImpl.java:546)
	at org.seasar.mayaa.impl.engine.EngineImpl.createTemplateInstance(EngineImpl.java:586)
	at org.seasar.mayaa.impl.engine.PageImpl.getTemplateFromFixedSuffix(PageImpl.java:148)
	at org.seasar.mayaa.impl.engine.PageImpl.getTemplate(PageImpl.java:162)
	at org.seasar.mayaa.impl.engine.RenderUtil.getTemplate(RenderUtil.java:299)
	at org.seasar.mayaa.impl.engine.RenderUtil.renderPage(RenderUtil.java:335)
	at org.seasar.mayaa.impl.engine.PageImpl.doPageRender(PageImpl.java:189)
	at org.seasar.mayaa.impl.engine.EngineImpl.doPageService(EngineImpl.java:373)
	at org.seasar.mayaa.impl.engine.EngineImpl.doService(EngineImpl.java:481)
      （中略）
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2170)
	... 54 more
```

--- a/src-impl/org/seasar/mayaa/impl/engine/processor/ElementProcessor.java
+++ b/src-impl/org/seasar/mayaa/impl/engine/processor/ElementProcessor.java
@@ -150,6 +150,10 @@ public class ElementProcessor extends AbstractAttributableProcessor
         return String.class;
     }
 
+    public Class<?> getInformalExpectedClass() {
+        return String.class;
+    }
+
     protected void resolvePrefix(PrefixAwareName name, Namespace currentNS) {
         if (name == null) {
             throw new IllegalArgumentException();

--- a/src/test/resources/it-case/engine/script/expected.html
+++ b/src/test/resources/it-case/engine/script/expected.html
@@ -25,7 +25,11 @@
 <span class="script5"></span>
 <span class="script6">6</span>
 <span class="script7">7</span>
-<span class="script8">8</span>
+<span data-value="8" class="script8">8</span>
+<!--  -->
+<span class="script9" data-value="9">9</span>
+<!--  -->
+<span class="script10" data-value="10">10</span>
 
 <dl id="numberList1">
 <dt id="numberList1numberKey0">10</dt><dd id="numberList1numberValue0">foo_b</dd>

--- a/src/test/resources/it-case/engine/script/target.html
+++ b/src/test/resources/it-case/engine/script/target.html
@@ -25,7 +25,11 @@
 <span class="script5">${ null }</span>
 <span class="script6">${ six }</span>
 <span class="script7">${ seven }</span>
-<span class="script8">${ eight }</span>
+<span class="script8" data-value="${ eight }">${ eight }</span>
+<!-- ${ var nine = 9; } -->
+<span class="script${nine}" data-value="${ nine }">${ nine }</span>
+<!-- ${ var ten = 9; ten++; ""} -->
+<span class="script${ten}" data-value="${ ten }">${ ten }</span>
 
 <dl id="numberList1">
 <dt id="numberKey">1</dt><dd id="numberValue">2</dd>

